app/src/main/java/it/quip/android/model/BaseParseObject.java

```diff
@@ -13,17 +11,13 @@ protected void safePut(String key, Object value) {
         }
     }
 
-    @Override
-    public String getObjectId() {
-        String objectId = super.getObjectId();
-        if (null == objectId) {
-            objectId = MockUtils.randomId();
-        }
-
-        return objectId;
-    }
-
     public void saveInternal() {
         this.saveInBackground();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        return ((ParseObject) o).getObjectId().equals(getObjectId());
```

Comment:
You might want to add a catch for a class cast exception here and return false if that happens